dpdispatcher/contexts/hdfs_context.py

```
@@ -151,100 +144,39 @@ def download(
-                if os.path.isdir(lfile):
-                    shutil.rmtree(lfile, ignore_errors=True)
-                elif os.path.isfile(lfile):
-                    os.remove(lfile)
-            shutil.move(rfile, lfile)
-
-        # remove tmp dir
-        shutil.rmtree(gz_dir, ignore_errors=True)
+            rfile_tgz = (
+                f"{self.remote_root}/{submission.submission_hash}_*_download.tar.gz"
+            )
+            HDFS.copy_to_local(rfile_tgz, f"{self.local_root}/tmp/")
```

Ruff (0.16.2)

[error] Probable insecure usage of temporary file or directory: "/tmp/"

(S108)